src/url.cpp

```diff
@@ -782,12 +782,20 @@ bool url::set_port(const std::string_view input) {
     port = std::nullopt;
     return true;
   }
+
+  auto non_digit_pos = std::find_if_not(trimmed.begin(), trimmed.end(),
```

Comment:
What if you check the first character, and if it is not a digit just return false?

Reply:
yess, that worked
```
  if (!ada::unicode::is_ascii_digit(trimmed.front())) {
    return false;
  }
```

Reply:
I added https://github.com/ada-url/ada/pull/839/commits/03d9cf9f804866c463ffb819cc5059c5bdc64070